plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostingFun/Core.hs

```diff
@@ -251,8 +275,9 @@ data ModelConstantOrTwoArguments = ModelConstantOrTwoArguments
 
 data ModelTwoArguments =
     ModelTwoArgumentsConstantCost       CostingInteger
-  | ModelTwoArgumentsLinearInX          ModelLinearSize
-  | ModelTwoArgumentsLinearInY          ModelLinearSize
+  | ModelTwoArgumentsLinearInX          OneVariableLinearFunction
```

Comment:
We can change the names of the things on the right without messing anything up, but changing the constructor names on the left changes the auto-derived JSON tags in the cost model file, which we probably don't want to do.

Reply:
It might be conceptually clearer if we renamed `ModelAddedSizes` and so on, but calling them all `Model...` helps to see that they're things of a similar type.  Another thing we could do would be to put these things in a separate file (`CostingFunction` or something) and use them in qualified form.

Reply:
I'd prefer if we renamed `ModelAddedSizes` and co too, otherwise with `OneVariableLinearFunction` it's all wildly inconsistent.

Reply:
I'll think about that a bit and maybe do it in another PR.  I think I might put these things in a separate file so that it's clear that they're all the same sort of thing.  There are a lot of things starting with `Model` and I was getting confused trying to sort them out mentally.